### PR TITLE
fix for first item in 4210 by doing labmda->lamb

### PR DESCRIPTION
**Pull request by eteq**
This fixes the issue pointed out in #4210 by simply renaming `lambda` -> `lamb` (the issue was that `lambda` is of course a reserved keyword in python.

### CODE REVIEW

**Comment by astrofrog** [comment]
@eteq - I wonder if ``lam`` would be a more 'common' abbreviation? Otherwise I'm thinking of the animal :-/

**Comment by mhvk** [comment]
The PEP8 way to deal with name conflicts with reserved keywords is to append an underscore, i.e., `lambda_`; from http://legacy.python.org/dev/peps/pep-0008/
"""
If your public attribute name collides with a reserved keyword, append a single trailing underscore to your attribute name. This is preferable to an abbreviation or corrupted spelling. (However, notwithstanding this rule, 'cls' is the preferred spelling for any variable or argument which is known to be a class, especially the first argument to a class method.)
"""

Though, personally, I'd prefer just `l`, `b`, as I never think of these as greek letters. Also, @barentsen mentions that is already used for `BarycentricTrueEcliptic`; shouldn't the two be consistent?!

**Comment by mhvk** [comment]
p.s.  Fix is for #4210 (not 4120); I went ahead and edited title and first entry.

**Comment by embray** [comment]
Incidentally, `astropy.modeling.projections` already uses "lam" as short for "lambda".  I'm not saying it's better or worse than "lamb" (I think I might like "lamb" better because it's cute :)  but maybe we should at least have a consistent way of abbreviating lambda: https://github.com/astropy/astropy/blob/master/astropy/modeling/projections.py#L721

I'd also be fine with "lambda_" per @mhvk's comment above, and would be willing to change the projection models to match it.  Anyways I don't have a strong feeling about which way we go as long as we try to keep some inter-subpackage consistency.

**Comment by astrofrog** [comment]
I'd vote for ``lam`` over ``lambda_``, and I think ``lam`` is a pretty common abbreviation, e.g. ``Flam``

**Comment by embray** [comment]
Can also always provide the unicode spelling as an alias :)

**Comment by barentsen** [comment]
In plain English I would always talk about ecliptic longitude and latitude, hence `lon` and `lat` is the first thing I intuitively tried when playing with the ecliptic coordinates.  Would this be an option?  (In this logic also `galactic` should have `lat` and `lon` arguments however.)

**Comment by mhvk** [comment]
Since not just `BarycentricTrueEcliptic` but also galactic uses `l` and `b`, it would seem most consistent to use `l` and `b` here too.

**Comment by eteq** [comment]
@mhvk @barentsen @astrofrog - just to clarify, I got these names because that's what I understand to be the convention among people who use ecliptic coordinates.  That is, `l`/`b`/`r` (or maybe `l`/`b`/`d`) is used for *heliocentric* ecliptic coordinates, and `lambda`/`beta`/`delta` are used for *geocentric* ecliptic.  My sources for that are https://en.wikipedia.org/wiki/Ecliptic_coordinate_system and an old textbook I found in the Yale astronomy library (which I think has since been shredded...)

**Comment by eteq** [comment]
But I strongly favor `lam` over `lambda_`.  I know about the convention, @mhvk, but I think most users will not, so they'll be more confused by it than an abbreviation.  I'd certainly use that if it was an internal variable, but for something public like this, I prefer `lam`.

All that said, it could be that we shouldn't care about the convention at all and just do `l`/`b`/`distance` for all of them.  Or even `lat`/`lon`/`distance`, as @barentsen suggested.  I don't care *that* much, but was just trying to do whatever seems to be most consistent with what people used to working with ecliptic coordinates might expect.

**Comment by eteq** [comment]
Oh, and FYI, @barentsen, you *can* always do ``coord.spherical.lon`` or  ``coord.spherical.lat``.  That's always available for any coordinate system - the conventional names are just there for what people are "used to".

**Comment by mhvk** [comment]
@eteq - OK, fair enough, I should have realised you had thought this through! Overall, I'd prefer to go with `lambda_, beta, delta`, then, following PEP8.

p.s. Note that the Nautical Almanac Office in planetary and lunar coordinates -- https://books.google.ca/books?id=2g_TICVzdSAC&pg=PR9&lpg=PR9&dq=geocentric+ecliptic+coordinates&source=bl&ots=CfyVxOk3Yq&sig=Jkx-UbBHXFu--dtubJqITbT72LM&hl=en&sa=X&ved=0CCIQ6AEwATgKahUKEwjm4vuTyqTIAhUCdj4KHdRXDBM#v=onepage&q=geocentric%20ecliptic%20coordinates&f=false -- seems to use `rho` for the distance, which seems a little more usual for a radial distance (but the astronomical almanac uses `pi` for the Moon; https://books.google.ca/books?id=ioogcZf033wC&pg=PA344&lpg=PA344&dq=geocentric+ecliptic+coordinates&source=bl&ots=0GOzanXvsY&sig=YILpG3z8QifgwCkwdfX1Q0Wb960&hl=en&sa=X&ved=0CC8Q6AEwBDgUahUKEwjayPnwyqTIAhXCcD4KHWIqDMs#v=onepage&q=geocentric%20ecliptic%20coordinates&f=false).

**Comment by mhvk** [comment]
OK, messages crossed. I also don't care all that much, but have a slight preference for just doing `l, b, distance` over adding abbreviations. But your call.

**Comment by eteq** [comment]
Honestly, what we really need is someone who actually knows the conventions people actually use... But awkwardly, it seems we can't get their feeback until we implement it... :/

At any rate, I tried to do a few more searches and concluded it's just a mess - lots of different conventions for what letter to use... In addition to what @mhvk dug up I found some refernces to `lambda`/`phi`/`delta`.  

So I guess now I tend towards `l`/`b`/`distance` just to make us internally consistent, or just changing all the ecliptic systems to `lat`/`lon`/`distance`.  Any strong opinions between those? (If no I'll just make a command decision)

**Comment by mhvk** [comment]
weak preference for `l, b`, by analogy with galactic. To me, longitude and latitude seem rather mixed in with locations on an object.

pinging JJ (@ijiraq) as someone who I think actually works with ecliptic systems in his Oort-cloud hunts.

**Comment by ijiraq** [comment]
Since @mhvk pinged me.  
I think I prefer  l, b, distance but I'm suspect many solar system people will wonder why galactic notation is being used for Ecliptic coordinates.  

As an example, pyEphem went through a period of using 'long' (oops) and then switched to lon/lat/distance     

Conclusion, lon/lat might be more recognized but l/b will not be confusing.  I do not like 'lam' or 'lamb' or 'lambda_'  

(how many characters in an attribute name is too many? , I guess 5).

**Comment by mtbannister** [comment]
I too am happier with lon/lat/distance. Confusion with Galactic notation is a real issue with l/b.

**Comment by astrofrog** [comment]
I would vote for lon/lat, since I don't think anyone will *complain* about that option.

**Comment by eteq** [comment]
Alright, sounds good - I'll change all of the ecliptic names to lon/lat/distance for consistency.

**Comment by astrofrog** [comment]
@eteq - better get in the change soon to leave Travis/Appveyor time to run before the feature freeze :)
